compiler/rustc_infer/src/infer/relate/higher_ranked.rs

```diff
@@ -26,8 +27,9 @@ impl<'tcx> InferCtxt<'tcx> {
     where
         T: TypeFoldable<TyCtxt<'tcx>>,
     {
-        if let Some(inner) = binder.clone().no_bound_vars() {
-            return inner;
+        // Inlined `no_bound_vars`.
+        if !binder.as_ref().skip_binder().has_escaping_bound_vars() {
+            return binder.skip_binder();
```

Comment:
to avoid clone?

do we do this more often, I've encountered this myself and have considered `try_no_bound_vars() -> Result<T, Binder<T>>` or maybe just changing `no_bound_vars` to that

Reply:
"do we do this more often"

as in: the pattern of `no_bound_vars()` and doing something with the binder if it returns `None`.

Reply:
ohhh, as far as i can tell, no this seems like the only place where we were previously repeatedly fighting the Copy bound bc no_bound_vars consumes its valur